tests/estimation/test_get_estimation.py

```diff
@@ -93,25 +78,107 @@
 ]
 
 
-@pytest.mark.parametrize("estimator", estimator_list)
-@pytest.mark.parametrize("config", [1, 5])
-@pytest.mark.parametrize("error_tolerance", [TOLERANCE])
-def test_tolerance(estimator, config, error_tolerance):
+parameter_name = [
+    "n",
+    "rg",
+    "mis_spec_m",
+    "mis_spec_y",
+    "dim_x",
+    "dim_m",
+    "seed",
+    "type_m",
+    "sigma_y",
+    "sigma_m",
+    "beta_t_factor",
+    "beta_m_factor",
+]
+
+
+parameter_list = list(
+    itertools.product(
+        [1000],
+        [default_rng(321)],
+        [False],
+        [False],
+        [1, 5],
+        [1],
+        [123],
+        ["binary"],
+        [0.5],
+        [0.5],
+        [0.5],
+        [0.5],
+    )
+)
+
+parameter_list.extend(
+    list(
+        itertools.product(
+            [1000],
+            [default_rng(321)],
+            [False],
+            [False],
+            [1, 5],
+            [1, 5],
+            [123],
+            ["continuous"],
+            [0.5],
+            [0.5],
+            [0.5],
+            [0.5],
+        )
+    )
+)
+
+
+@pytest.fixture(params=parameter_list)
+def dict_param(request):
+    return dict(zip(parameter_name, request.param))
+
+
+@pytest.fixture(autouse=True)
+def data_simulation(dict_param):
+    res = simulate_data(**dict_param)
+    global x, t, m, y, effects
+    x = res[0]
+    t = res[1].ravel()
+    m = res[2]
+    y = res[3].ravel()
+    effects = np.array(res[4:9])
+
+
+@pytest.fixture(params=estimator_list, autouse=True)
+def skip_if_not_implemented(request, dict_param):
+    # config determination
+    if dict_param["dim_m"] == 1 and dict_param["type_m"] == "binary":
+        config = 0
+    else:
+        config = 5
+
+    # try whether estimator is implemented or not
     try:
-        effects_chap = get_estimation(x, t.ravel(), m, y.ravel(), estimator, config)
-        # effects_chap = get_estimation(x, t, m, y, estimator, config)
-        effects_chap = effects_chap[0:5]
-    except Exception as get_estimation_error:
-        if get_estimation_error != NotImplementedError:
+        global effects_chap
+        effects_chap = get_estimation(x, t, m, y, request.param, config)[0:5]
+    except Exception as message_error:
+        if message_error != NotImplementedError:
             pytest.xfail("Missing NotImplementedError")
         else:
             pytest.mark.skip("Not implemented")
-    else:
-        error = abs((effects_chap - effects) / effects)
-        if np.all(np.isnan(effects_chap)):
-            pytest.skip("all effects are NaN")
-        elif np.any(np.isnan(effects_chap)):
-            pprint("NaN found")
-            assert np.all(error[~np.isnan(error)] <= error_tolerance[~np.isnan(error)])
-        else:
-            assert np.all(error <= error_tolerance)
+
+    # NaN situations
```

Comment:
This should never occur ?

Reply:
No it occurs. For now, non-implemented or half-implemented estimators return some NaN.

Reply:
The point is that we should distinguish between an estimation procedure that failed for numerical reason, and anon-implemented thing. 
Isn't it safer to return 0 when the estimator is not yet implemented ?

Reply:
returning 0 might let the user think that the estimated effect is zero, which is more confusing IMO. For the multiply robust estimator I plan to include a more explicit warning in the doc (issue #17)
for the estimators that are not implemented in case the mediator is a not a 1D binary raise an error rather than returning NaNs? a ValueError to state that it the statistical estimator does not support this case rather than the idea it's not implemented yet but will be later?

Reply:
Ok, I am going to implement those ValueErrors.

Reply:
Now it xfails if all effects are NaN, 36f205a

I opened #22 to remember putting ValueErrors in `get_estimation.py`